modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBString.java

```diff
@@ -67,11 +74,12 @@ public Expression stringDowncase(){
      * @return
      */
     public Expression stringUpcase(){
-        if (json.get(1) instanceof String){
-            String s = (String) json.get(1);
-            return ff.literal(s.toUpperCase());
+        if (parse.get(json,1) instanceof String){
+            String s =  parse.get(json, 1);
+            Function function = ff.function("strToUpperCase", ff.literal(s));
+            return ff.literal(function.evaluate(s));
         } else {
-            return ff.literal("Requires instance of string to convert to uppercase");
+            throw new MBFormatException("Requires instance of string to convert to uppercase");
```

Comment:
This looks like it wouldn't support `["upcase", ["concat", "c", "a", "t"]]`, which seems to be an error.